### PR TITLE
ARK 224 Additions

### PR DESCRIPTION
**Pull request by Bletch1971**
1. Added the Compy.
2. Added the MagnifyingGlass, RocketTurret, SeaMine and WallTorch
   engrams.
3. Added ServerAdminLogs and BufferSize to Administration - Server
   Options section.
4. Added AllowCustomRecipes to Rules section.
5. Added PassiveDefensesDamageRiderlessDinos to Structures section.

Note: GameData.cs file had a lot of Engram changes. This was due to
adding an extra space to allow for a 3 digit EngramPointsCost value.

### CODE REVIEW

**Comment by james-d-elliott** [comment]
Damn, beat me to it. Have some other changes to submit though. c;

File: ARK Server Manager/Lib/Model/ServerProfile.cs [hunk 1]
```diff
@@ -1676,10 +1713,7 @@ public string GetServerArgs()
                 serverArgs.Append("?ForceAllStructureLocking=true");
             }
 
-            if (this.AutoDestroyOldStructuresMultiplier >= 1.0f)
-            {
-                serverArgs.AppendFormat("?AutoDestroyOldStructuresMultiplier={0}", AutoDestroyOldStructuresMultiplier);
-            }
+            serverArgs.AppendFormat("?AutoDestroyOldStructuresMultiplier={0}", AutoDestroyOldStructuresMultiplier);
```
Comment (ChronosWS) [inline ARK Server Manager/Lib/Model/ServerProfile.cs]: Consider string interpolation:
.Append($"?AutoDestroyOldStructuresMultiplier={AutoDestroyOldStructuresMultiplier}") 

I find it slightly easier to read than String.Format() style composition.

File: ARK Server Manager/ServerSettingsControl.xaml [hunk 2]
```diff
@@ -905,17 +905,22 @@
                             <GroupBox Grid.Row="10" Grid.ColumnSpan="6" Header="Server Options">
                                 <Grid>
                                     <Grid.ColumnDefinitions>
-                                        <ColumnDefinition/>
-                                        <ColumnDefinition/>
+                                        <ColumnDefinition Width="67*"/>
```
Comment (ChronosWS) [inline ARK Server Manager/ServerSettingsControl.xaml]: Did you mean to use the \* notation here, which sets relative widths, or do you intend absolute widths?

**Comment by Bletch1971** [comment]
Thank you for the heads up.

**Comment by lstanden** [comment]
Need to add the new level progression 89-92 as well?

**Comment by Bletch1971** [comment]
Currently the wiki does not have the details for the three new levels. I would love to add the information but cannot find any source that has it.

Also the new levels will only be used if the ASM user resets the player/dino levels back to the official ones using the button. If the ASM user does not use the custom levels, then it reverts back to the ARK defaults.

Do you have a source for the new level details?

**Comment by james-d-elliott** [comment]
Max level is now like 94. They constantly make mistakes when doing this. I tested on a vanilla install for development/testing. I was just going to pull all this after but it's probably easier to get the changes done all at once.

Creatures (new max XP is..well I think it's exactly 225000):

```
            new Level { XPRequired=98000,  EngramPoints=0 },
            new Level { XPRequired=120000, EngramPoints=0 },
            new Level { XPRequired=150000, EngramPoints=0 },
            new Level { XPRequired=185000, EngramPoints=0 },
            new Level { XPRequired=225000, EngramPoints=0 },
```

Survivors (new max XP is 873540):

```
            new Level { XPRequired=506186, EngramPoints=60 },
            new Level { XPRequired=566358, EngramPoints=60 },
            new Level { XPRequired=632547, EngramPoints=60 },
            new Level { XPRequired=705354, EngramPoints=60 },
            new Level { XPRequired=785441, EngramPoints=60 },
            new Level { XPRequired=873538, EngramPoints=60 },
```

**Comment by Bletch1971** [comment]
Strange, I got the same information when I setup a test world. Apparently there is only 3 new levels, but I get the same as you.
Not sure why there is a difference with the information and the game.

Edit - I got one more dino level than you did
new Level {XPRequired=275000 , EngramPoints=0},

**Comment by james-d-elliott** [comment]
@Bletch1971 that's just the new XP max, if you check again you don't actually get a level when you get there.

**Pull request merged**